Panic: setRateLimitXResponseHeaders

# Problem encountered
One data-node on Devnet stopped due to the panic

# Observed behaviour
Node stopped

# Expected behaviour
Node should not stop

# Evidence

## Logs
```log
2023-02-15T02:11:00.277Z        ERROR        datanode.api.grpc        api/server.go:324        failed to set header        {"error": "rpc error: code = Internal desc = transport: SendHeader called multiple times"}
2023-02-15T02:11:00.280Z        ERROR        datanode.api.grpc        api/server.go:324        failed to set header        {"error": "rpc error: code = Internal desc = transport: SendHeader called multiple times"}
2023-02-15T02:11:00.282Z        ERROR        datanode.api.grpc        api/server.go:324        failed to set header        {"error": "rpc error: code = Internal desc = transport: SendHeader called multiple times"}
2023-02-15T02:11:00.282Z        ERROR        datanode.api.grpc        api/server.go:324        failed to set header        {"error": "rpc error: code = Internal desc = transport: SendHeader called multiple times"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2cede48]
goroutine 896023 [running]:
code.vegaprotocol.io/vega/datanode/ratelimit.setRateLimitXResponseHeaders({0x531d6d0, 0xc047d0e5a0}, 0x0, 0x13?, {0xc0472797d0, 0xb})
        /jenkins/workspace/private/Deployments/Publish-vega-dev-releases/vega/datanode/ratelimit/ratelimit.go:179 +0x2e8
code.vegaprotocol.io/vega/datanode/ratelimit.(*RateLimit).GRPCInterceptor(0xc005f00540, {0x531d6d0, 0xc047d0e5a0}, {0x4674a20, 0xc047d0e540}, 0x5a3ecf3?, 0xc047d7a240)
        /jenkins/workspace/private/Deployments/Publish-vega-dev-releases/vega/datanode/ratelimit/ratelimit.go:158 +0x2e5
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x531d6d0?, 0xc047d0e5a0?}, {0x4674a20?, 0xc047d0e540?})
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1162 +0x5b
code.vegaprotocol.io/vega/datanode/api.headersInterceptor.func1({0x531d6d0, 0xc047d0e5a0}, {0x4674a20, 0xc047d0e540}, 0x42c5100?, 0xc0581ebf40)
        /jenkins/workspace/private/Deployments/Publish-vega-dev-releases/vega/datanode/api/server.go:328 +0x5e3
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x531d6d0?, 0xc047d0e5a0?}, {0x4674a20?, 0xc047d0e540?})
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1165 +0x83
code.vegaprotocol.io/vega/datanode/api.remoteAddrInterceptor.func1({0x531d6d0, 0xc047d0e510}, {0x4674a20, 0xc047d0e540}, 0xc047d768c0, 0xc0581ebf40)
        /jenkins/workspace/private/Deployments/Publish-vega-dev-releases/vega/datanode/api/server.go:281 +0x637
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x531d6d0?, 0xc047d0e510?}, {0x4674a20?, 0xc047d0e540?})
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1165 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1({0x531d6d0, 0xc047d0e510}, {0x4674a20, 0xc047d0e540}, 0xc047d768c0, 0xc047d7a240)
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1167 +0x12b
code.vegaprotocol.io/vega/protos/vega/api/v1._CoreService_Statistics_Handler({0x4864980?, 0xc001aac280}, {0x531d6d0, 0xc047d0e510}, 0xc04722b650, 0xc005f005c0)
        /jenkins/workspace/private/Deployments/Publish-vega-dev-releases/vega/protos/vega/api/v1/core_grpc.pb.go:332 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc005f0e960, {0x5335c60, 0xc0475949c0}, 0xc0474445a0, 0xc003897260, 0x6e687f0, 0x0)
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc005f0e960, {0x5335c60, 0xc0475949c0}, 0xc0474445a0, 0x0)
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /jenkins/GOPATH/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:963 +0x28a
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -69,6 +69,7 @@
 - [7492](https://github.com/vegaprotocol/vega/issues/7492) - Send market depth events at the end of each block
 - [7582](https://github.com/vegaprotocol/vega/issues/7582) - Validate transfer amount in `checkTx`
 - [7577](https://github.com/vegaprotocol/vega/issues/7577) - Use correct trade size when calculating pending open volume
+- [7598](https://github.com/vegaprotocol/vega/issues/7598) - Set up log for rate limiter
 
 ## 0.67.2
 

--- a/datanode/ratelimit/ratelimit.go
+++ b/datanode/ratelimit/ratelimit.go
@@ -62,6 +62,7 @@ func NewFromConfig(cfg *Config, log *logging.Logger) *RateLimit {
 	r := &RateLimit{
 		lmt:         lmt,
 		naughtyStep: ns,
+		log:         log,
 	}
 	r.cfg.Store(cfg)
 	return r

